packages/tizen_rpc_port/example/server/lib/message_server.dart

```diff
@@ -0,0 +1,205 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: public_member_api_docs
+
+import 'package:tizen_rpc_port/tizen_rpc_port.dart';
+
+const String _tidlVersion = '1.9.1';
+
+enum _DelegateId {
+  notifyCallback(1);
+
+  const _DelegateId(this.id);
+  final int id;
+}
+
+enum _MethodId {
+  result(0),
+  callback(1),
+  register(2),
+  unregister(3),
+  send(4);
+
+  const _MethodId(this.id);
+  final int id;
+}
+
+abstract class _CallbackBase extends Parcelable {
+  _CallbackBase(this.id, this.once) {
+    sequenceId = sequenceNum++;
+  }
+
+  int id = 0;
+  bool once = false;
+  int sequenceId = 0;
+  static int sequenceNum = 0;
+
+  String get tag => '$id::$sequenceId';
+
+  @override
+  void serialize(Parcel parcel) {
+    parcel.writeInt32(id);
+    parcel.writeInt32(sequenceId);
+    parcel.writeBool(once);
+  }
+
+  @override
+  void deserialize(Parcel parcel) {
+    id = parcel.readInt32();
+    sequenceId = parcel.readInt32();
+    once = parcel.readBool();
+  }
+}
+
+abstract class ServiceBase {
+  ServiceBase(this.sender, this.instance);
+
+  final String sender;
+  final String instance;
+
+  Port? _port;
+
+  Future<void> disconnect() async {
+    _port?.disconnect();
+    _port = null;
+  }
+
+  Future<void> onCreate();
+
+  Future<void> onTerminate();
+
+  Future<int> onRegister(String name, NotifyCallback callback);
+
+  Future<void> onUnregister();
+
+  Future<int> onSend(String message);
+}
+
+class NotifyCallback extends _CallbackBase {
+  NotifyCallback(this._port, this.service, {bool once = false})
+      : super(_DelegateId.notifyCallback.id, once);
+  final Port? _port;
+  ServiceBase service;
+  bool _valid = true;
+
+  Future<void> invoke(String sender, String message) async {
+    if (_port == null) {
+      throw StateError('Must be connected first');
+    }
+
+    if (once && !_valid) {
+      throw Exception('InvalidCallbackException');
+    }
+
+    final Parcel parcel = Parcel();
+    parcel.writeInt32(_MethodId.callback.id);
+    serialize(parcel);
+    parcel.writeString(sender);
+    parcel.writeString(message);
+    if (_port != null) {
+      parcel.send(_port!);
+    }
+
+    _valid = false;
+  }
+}
+
+abstract class Message extends StubBase {
+  Message() : super('Message') {
+    _methodHandlers[_MethodId.register.id] = _onRegisterMethod;
+    _methodHandlers[_MethodId.unregister.id] = _onUnregisterMethod;
+    _methodHandlers[_MethodId.send.id] = _onSendMethod;
+  }
+
+  List<ServiceBase> services = <ServiceBase>[];
+  final Map<int, dynamic> _methodHandlers = <int, dynamic>{};
+
+  @override
+  Future<void> listen() async {
+    return await super.listen();
+  }
```

Comment:
Done. Thanks!
But I have a question.
As you said, the logic is unnecessary, but our intention was to specify the methods needed for the interface so that app developers could easily know what functions they have.
As you said, wouldn't it be difficult for app developers to recognize the API above if they were omitted?

Reply:
Assuming that the developers use the Dart extension and the Dart analyzer integrated within it, there should be no such problem. (Try pressing `Ctrl`+`.` after placing your cursor after `_server.` in `initState`.)

Reply:
Ok. Thanks for the reply!